src/service/context/mod.rs

```diff
@@ -152,6 +152,50 @@ impl<S> Context<S> {
         self.extensions.get::<T>()
     }
 
+    /// Get a reference to an extension or T's [default()](std::default::Default).
+    /// Refer to [Context::get](Self::get) for more detail's
+    /// # Example
+    /// ```
+    /// # use rama::service::Context;
+    /// # let mut ctx = Context::default();
+    /// # ctx.insert(5i32);
+    ///
+    /// // Test existing value
+    /// assert_eq!(ctx.get_or_default::<i32>(), &5i32);
+    ///
+    /// assert_eq!(ctx.get_or_default::<f64>(), &0f64);
+    /// ```
+    pub fn get_or_default<T: Send + Default + Clone + Sync + 'static>(&mut self) -> &T {
+        if !self.extensions.contains::<T>() {
```

Comment:
Again, this should be implemented directly in `extensions`. Do think you should first try to use `.get(...)` and drop the contains concept. The `get` public method tries to also fetch from a parent extension which is still a concept I am supporting. Might in the end never the less drop it, but for now it is there, and if possible I'll keep it. But contains is not something we need to expose and currently anyway not support the parent concept if I see your approach to it.